### PR TITLE
SASL implementation. 

### PR DESCRIPTION
**Pull request by nolsen42**
This implementation contains working SASL PLAIN authentication.

### CODE REVIEW

**Comment by nolsen42** [comment]
It took a bit of commits to get it right, but commit 9af6801 and aeea781 are the final fixes.

File: config.json.example [hunk 1]
```diff
@@ -4,6 +4,9 @@
   "ident": "conejo",
   "gecos": "Soy amigable",
   "password": "",
+  "sasl": "False",
```
Comment (ghost) [inline config.json.example]: This should be `false`.

**Comment by nolsen42** [comment]
Fixed.

**Comment by nolsen42** [comment]
Done.

**Comment by Polsaker** [comment]
Well, using `== True` is dumb, but fuck this ¯\_(ツ)_/¯

**Pull request merged**